cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocRuntimeException.java

```diff
@@ -16,78 +16,264 @@
 
 package io.cdap.cdap.runtime.spi.provisioner.dataproc;
 
+import com.google.api.gax.rpc.ApiException;
 import com.google.common.base.Strings;
-import com.google.common.base.Throwables;
-import io.cdap.cdap.error.api.ErrorTagProvider;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.HashSet;
-import java.util.Set;
+import io.cdap.cdap.api.exception.ErrorCategory;
+import io.cdap.cdap.api.exception.ErrorCategory.ErrorCategoryEnum;
+import io.cdap.cdap.api.exception.ErrorCodeType;
+import io.cdap.cdap.api.exception.ErrorType;
+import io.cdap.cdap.api.exception.FailureDetailsProvider;
 import javax.annotation.Nullable;
 
 /**
  * A {@link RuntimeException} that wraps exceptions from Dataproc operation and provide a {@link
  * #toString()} implementation that doesn't include this exception class name and with the root
  * cause error message.
  */
-public class DataprocRuntimeException extends RuntimeException implements ErrorTagProvider {
+public class DataprocRuntimeException extends RuntimeException implements FailureDetailsProvider {
 
-  private final Set<ErrorTag> errorTags = new HashSet<>();
+  private static final String TROUBLESHOOTING_MESSAGE_FORMAT =
+    "For troubleshooting Dataproc errors, refer to %s";
+  public static final String TROUBLESHOOTING_DOC_URL =
+      "https://cloud.google.com/dataproc/docs/support/troubleshoot-cluster-creation";
+  public static final ErrorCategory ERROR_CATEGORY_CONFIGURATION =
```

Comment:
Name it ERROR_CATEGORY_PROVISIONING_CONFIGURATION to ensure it's not used in other scenarios

Reply:
done.